lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -37,40 +37,42 @@ using namespace swift;
 /************************ PROPERTY SYNTHESIS **********************************/
 /******************************************************************************/
 
-static VarDecl*
- lookupDistributedActorProperty(NominalTypeDecl *decl, DeclName name) {
-   assert(decl && "decl was null");
-   auto &C = decl->getASTContext();
-
-   auto clazz = dyn_cast<ClassDecl>(decl);
-   if (!clazz)
-     return nullptr;
-
-   auto refs = decl->lookupDirect(name);
-   if (refs.size() != 1)
-     return nullptr;
-
-   auto var = dyn_cast<VarDecl>(refs.front());
-   if (!var)
-     return nullptr;
-
-   Type expectedType = Type();
-   if (name == C.Id_id) {
-     expectedType = getDistributedActorIDType(decl);
-   } else if (name == C.Id_actorSystem) {
-     expectedType = getDistributedActorSystemType(decl);
-   } else {
-     llvm_unreachable("Unexpected distributed actor property lookup!");
-   }
-   if (!expectedType)
-     return nullptr;
-
-   if (!var->getInterfaceType()->isEqual(expectedType))
-     return nullptr;
-
-   assert(var->isSynthesized() && "Expected compiler synthesized property");
-   return var;
- }
+static VarDecl *lookupDistributedActorProperty(NominalTypeDecl *decl,
+                                               DeclName name) {
+  assert(decl && "decl was null");
+  auto &C = decl->getASTContext();
+
+  auto clazz = dyn_cast<ClassDecl>(decl);
+  if (!clazz)
+    return nullptr;
+
+  auto refs = decl->lookupDirect(name);
+  if (refs.size() != 1)
```

Comment:
This is going to break if, say, some extension somewhere adds an unrelated function with the same name. Usually we walk the lookup results to find what we're looking for, ignoring anything with the same name.

It doesn't have to be addressed right now, but perhaps in the future.

Reply:
Ah right, going to fix it.